### PR TITLE
Front door security rules IAC fix

### PR DESCRIPTION
**Pull request by dborgen2**
## What’s changing?

PR [3147](https://github.com/18F/piipan/pull/3147) had an issue when publishing for the first time, due to this issue:[https://github.com/MicrosoftDocs/azure-docs/issues/61497](https://github.com/MicrosoftDocs/azure-docs/issues/61497)

To get around that issue, we need to create the rules engine initially, but then associate it to the front door routing rule after front door was created.

## Why?

Follow-up to PR [3147](https://github.com/18F/piipan/pull/3147)

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [x] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by ryangeorgeharvey** [comment]
@dborgen2 Have you tested these IAC changes for both a new provision and re-provision?

**Comment by dborgen2** [comment]
> @dborgen2 Have you tested these IAC changes for both a new provision and re-provision?

I did. I deleted my entire front door and then ran it, and then ran it again and both times it seemed to work.

File: iac/add-front-door-to-app.bash [hunk 1]
```diff
@@ -39,7 +40,11 @@ main () {
       resourceTags="$RESOURCE_TAGS" \
       wafPolicyName="$waf_name" \
       prefix="$PREFIX" \
-      env="$ENV"
+      env="$ENV" \
+      rulesEngineName="$rules_engine_name"
+
+  echo 'Associate Routing Rule to Front Door Rule Engine'
+  az network front-door routing-rule update --front-door-name "$front_door_name" --name routingRule1 --resource-group "$resource_group" --rules-engine "$rules_engine_name"
```
Comment (ryangeorgeharvey) [inline iac/add-front-door-to-app.bash:47]: Nitpick (non-blocking): I don't like the hardcoded use of `routingRule1`. You could used `az network front-door routing-rule list` and a query to get the exact rule, but I'm not sure how you would differentiate the rule names in the result.

**Review by ryangeorgeharvey — APPROVED** [review]
Solution isn't ideal, but I confirmed this is a current bug with ARM. Based on that, this LGTM.

**Review by bgfoshay — APPROVED** [review]
LGTM 👍

File: iac/add-front-door-to-app.bash (hunk 1 above)
Comment (dborgen2) [reply iac/add-front-door-to-app.bash:47]: Yea, good point. Perhaps something to look into in the future

**Pull request merged**